tests/sandbox/test_jax.py

```diff
@@ -816,6 +816,78 @@ def test_second():
     compare_jax_and_py(fgraph, [np.zeros([5], dtype=theano.config.floatX), 5.0])
 
 
+def test_jax_BatchedDot():
+    # tensor3 . tensor3
+    a = tt.tensor3("a")
+    a.tag.test_value = (
+        np.linspace(-1, 1, 10 * 5 * 3).astype(theano.config.floatX).reshape((10, 5, 3))
+    )
+    b = tt.tensor3("b")
+    b.tag.test_value = (
+        np.linspace(1, -1, 10 * 3 * 2).astype(theano.config.floatX).reshape((10, 3, 2))
+    )
+    out = tt.blas.BatchedDot()(a, b)
+    fgraph = theano.gof.FunctionGraph([a, b], [out])
+    compare_jax_and_py(fgraph, [get_test_value(i) for i in fgraph.inputs])
+
+    # matrix . matrix
+    a = tt.matrix("a")
+    a.tag.test_value = (
+        np.linspace(-1, 1, 5 * 3).astype(theano.config.floatX).reshape((5, 3))
+    )
+    b = tt.matrix("b")
+    b.tag.test_value = (
+        np.linspace(1, -1, 5 * 3).astype(theano.config.floatX).reshape((5, 3))
+    )
+    out = tt.blas.BatchedDot()(a, b)
+    fgraph = theano.gof.FunctionGraph([a, b], [out])
+    compare_jax_and_py(fgraph, [get_test_value(i) for i in fgraph.inputs])
+
+    # matrix . tensor3
+    a = tt.matrix("a")
+    a.tag.test_value = (
+        np.linspace(-1, 1, 10 * 5 * 1).astype(theano.config.floatX).reshape((10, 5))
+    )
+    b = tt.tensor3("b")
+    b.tag.test_value = (
+        np.linspace(1, -1, 10 * 5 * 2).astype(theano.config.floatX).reshape((10, 5, 2))
+    )
+    out = tt.blas.BatchedDot()(a, b)
+    fgraph = theano.gof.FunctionGraph([a, b], [out])
+    compare_jax_and_py(fgraph, [get_test_value(i) for i in fgraph.inputs])
+
+    # tensor3 . matrix
+    a = tt.tensor3("a")
+    a.tag.test_value = (
+        np.linspace(-1, 1, 10 * 5 * 3).astype(theano.config.floatX).reshape((10, 5, 3))
+    )
+    b = tt.matrix("b")
+    b.tag.test_value = (
+        np.linspace(1, -1, 10 * 3).astype(theano.config.floatX).reshape((10, 3))
+    )
+    out = tt.blas.BatchedDot()(a, b)
+    fgraph = theano.gof.FunctionGraph([a, b], [out])
+    compare_jax_and_py(fgraph, [get_test_value(i) for i in fgraph.inputs])
```

Comment:
All these tests are only hitting one of two independent branch paths in `jax_funcify_BatchedDot`.  Unless we're unsure about the functionality of `jnp.einsum` over time, we don't want to add tests for it; otherwise, our already slow test suite will get much, much slower.

Reply:
Fair - I removed two of these permutations.